### PR TITLE
Add allergies exercise

### PR DESCRIPTION
**Pull request by jonmcalder**
Add example solution and tests
Add exercise metadata (UUID, difficulty, topics, unlocked_by etc)
Generate README

### CODE REVIEW

File: exercises/allergies/example.R [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    chocolate = 32,
+    tomatoes = 16,
+    strawberries = 8,
+    shellfish = 4,
+    peanuts = 2,
+    eggs = 1
+  )
+  
+  check_allergy <- function(allergy, val) {
+    if (val %% (allergy*2) >= allergy) {
```
Comment (katrinleinweber) [inline exercises/allergies/example.R]: Shouldn't the linter have found the missing spaces around `*`? ;-)

Comment (jonmcalder) [inline exercises/allergies/example.R]: Well spotted! Turns out I didn't actually include that check in the linter. I should probably add it though. Will do a separate PR for that.

**Comment by jonmcalder** [comment]
Ok no problem - @lorenzwalthert has agreed to have a look at it for us later this month when he's got some time.

File: exercises/allergies/example.R [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    chocolate = 32,
+    tomatoes = 16,
+    strawberries = 8,
+    shellfish = 4,
+    peanuts = 2,
+    eggs = 1
+  )
+  
+  check_allergy <- function(allergy, val) {
+    val %% (allergy * 2) >= allergy
```
Comment (katrinleinweber) [inline exercises/allergies/example.R]: Not a _review comment_, but a good old comprehension question: Is this line a "greedy" check whether any of the given allergy values fits into the total score? In a C tutorial over a year ago, I implemented a greedy algorithm to calculate composing a given amount of change (money) from as few as possible, as large as possible coins. The problem here kind of matches, but I'm having a hard time understanding why this line of math would solve the problem. It does, though O.o

File: exercises/allergies/example.R [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "cats"
+  )
+  
+  check_allergy <- function(allergy, val) {
+    intToBits(val)[which(allergy_list == allergy)] %>% as.logical()
+  }
+  
+  allergies <- sapply(allergy_list, check_allergy, num)
+  
+  structure(allergy_list[allergies], class = 'allergy')
```
Comment (lorenzwalthert) [inline exercises/allergies/example.R]: use double quotes

Comment (jonmcalder) [inline exercises/allergies/example.R]: 👍
